tools/toolchains/__init__.py

```diff
@@ -188,6 +188,23 @@ def __str__(self):
 }
 
 
+def check_toolchain_path(function):
+    """Check if the path to toolchain is valid. Exit if not.
+    Use this function as a decorator.  Causes a system exit if the path does
+    not exist. Execute the function as normal if the path does exist.
+
+    Positional arguments:
+    function -- the function to decorate
+    """
+    def perform_check(self, *args, **kwargs):
+        if not exists(self.toolchain_path):
+            print('[ERROR] Toolchain path for %s does not exist.\n'
+                  'Current value: %s' % (self.name, self.toolchain_path))
+            sys.exit(-1)
```

Comment:
Thats good starting point 👍  . Somebody should look about existing return codes, standardize all of them and prepare proper documentation..